scripts/seedCache.js

```diff
@@ -0,0 +1,54 @@
+import axios from "axios";
+import log from "npmlog";
+
+const PORT = Number(process.env.PORT || 3000);
+
+const axiosInstance = axios.create({
+  baseURL: `http://localhost:${PORT}`,
+});
+
+(async () => {
+  if (process.env.TRAVIS !== undefined) {
+    await axiosInstance.post("/ready");
+
+    return;
+  }
+
+  // ==================================================
+  // Agreements
+
+  const { data: indexedAgreements } = await axiosInstance.get("/agreements");
+  const indexedAgreementsLength = indexedAgreements.length;
+  let index = -1;
+  while (++index < indexedAgreementsLength) {
+    const { id, num } = indexedAgreements[index];
+    const idcc = String(num).padStart(4, "0");
+
+    log.info(
+      "scripts/seedCache.js",
+      `[Agreement ${index + 1}/${indexedAgreementsLength}] Seeding cache for ID=${id}…`,
+    );
+    await axiosInstance.get(`/agreement/articles?agreementIdOrIdcc=${id}&query=1`);
+    await axiosInstance.get(`/agreement/articles?agreementIdOrIdcc=${idcc}&query=1`);
+  }
+
+  // ==================================================
+  // Codes
+
+  const { data: indexedCodes } = await axiosInstance.get("/codes");
+  const filteredIndexedCodes = indexedCodes.filter(({ id }) => id === "LEGITEXT000006072050");
+  const indexedCodesLength = filteredIndexedCodes.length;
+  index = -1;
+  while (++index < indexedCodesLength) {
+    const { id } = filteredIndexedCodes[index];
+
+    log.info(
+      "scripts/seedCache.js",
+      `[Code ${index + 1}/${indexedCodesLength}] Seeding cache for ID=${id}…`,
+    );
+    await axiosInstance.get(`/code/articles?codeId=${id}&query=1`);
```

Comment:
also
```
for(codeId of filteredIndexedCodes) {
   await axiosInstance.get(`/code/articles?codeId=${id}&query=1`);
}
```  
may be simpler since you don't really need index

Reply:
Hm I'm not sure parallel calls is possible since it a very intensive task (to enrich articles with their `path`) unless we use multi-threading. At least locally, it's damn heavy for me when run in band (the api is unreachable during the cache seeding) and I have a Threadripper 32...